### PR TITLE
module: optimize the path searching

### PR DESCRIPTION
**Pull request by yorkie**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included(modified)
- [x] documentation is changed or added

Give the following optimizations:

- search rules:
  - If the `id` is a relative path, directly search by `__dirname`, no global module and `node_modules`.
  - If the `id` is a global path, directly try with file.
- tweak the order of global paths, level up the `NODE_PATH`.
- remove the retry with `index` after package main property is absence, because checked already before.

Also, this patch removes the way to load relative module without dots:

```js
require('test/foobar') // doesn't work
require('./test/foobar') // it's working
```

### CODE REVIEW

**Comment by yorkie** [comment]
This fixes #262, too.

**Comment by legendecas** [comment]
IMHO, use `NODE_PRIORITIZED_PATH` environ to set a high priority search path rather than reversing search path order might be better

**Comment by yorkie** [comment]
> IMHO, use NODE_PRIORITIZED_PATH environ to set a high priority search path rather than reversing search path order might be better

Neat! I will change this.

**Comment by yorkie** [comment]
Added NODE_PRIORITIZED_PATH, @legendecas :)

File: docs/api/Module.md [hunk 1]
```diff
@@ -29,10 +29,10 @@ If a native module named `id` exists, load it and return.
 
 `require` function searches for modules in the following order:
 
-1. Current working directory.
-2. `iotjs_modules` folder under current working directory.
-3. `$HOME/iotjs_modules`
-4. `$IOTJS_PATH/iotjs_modules`
+1. `$NODE_PRIORITIZED_PATH/node_modules` if present.
+1. `node_modules` folder under current working directory.
+2. `$NODE_PATH/node_modules`.
```
Comment (legendecas) [inline docs/api/Module.md]: Actually, if a `NODE_PATH` which is current working directory, relative js file can be loaded without leading `./`.

e.g. with following directory structure:

```
- /home/workspace
|- lib
|    |- foobar.js
|- index.js
```

and set `NODE_PATH` to `/home/workspace`,

and with following content in `index.js`, it works:

```js
require('lib/foobar')
```

So it should be `2. $NODE_PATH`

Comment (yorkie) [reply docs/api/Module.md]: The above example doesn't work at IoT.js and ShadowNode. The search path for NODE_PATH should be `$NODE_PATH/node_modules`.

Comment (legendecas) [reply docs/api/Module.md]: It works in Node.js, should we support it too?

**Comment by yorkie** [comment]
Remove the node_modules postfix after NODE_PATH and NODE_PRIORITIZED_PATH @legendecas :)

**Pull request merged**